### ISSUE TITLE
f-strings disable pylint file directives

### ISSUE THREAD

**Issue by jacebrowning** [PyCQA/pylint#1371]
### Steps to reproduce

Create two python files:

```python
# pylint: disable=unused-variable

def sample(x):
    test = "{x}".format(x=x)
```

```python
# pylint: disable=unused-variable

def sample(x):
    test = f"{x}"
```

Run both through pylint.

### Current behavior

The second file reports `W0612: Unused variable 'test'`.

### Expected behavior

Neither file reports an unused variable because the rule is disabled for the file.

### pylint --version output

I'm running the cutting edge to test Python 3 support:

```
No config file found, using default configuration
pylint 2.0.0,
astroid 1.5.0
Python 3.6.0 (default, Dec 27 2016, 18:35:41)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

```
[dev-packages.pylint]
git = "https://github.com/PyCQA/pylint.git"
ref = "89d766ef8bea922cfacf55f8a8adbbf783955d4a"

[dev-packages.astroid]
git = "https://github.com/PyCQA/astroid.git"
ref = "dc7cdd223f7d0897d7c5f6c3dd89143ca3128bc7"
```

**Comment by PCManticore** [comment]
Thank you, I can reproduce this issue.

**Comment by PCManticore** [comment]
I forgot about this issuehile creating this another one: https://github.com/PyCQA/pylint/issues/1373. In short, it is caused by a CPython bug. It should be solved on our side with the latest astroid changes.
